_drafts/ember-orbit.md

```diff
@@ -0,0 +1,477 @@
+});
+```
+
+We access `this.store` as usual, but now instead of an Ember Data store it’s an Ember-Orbit store. So instead of calling Ember Data’s `findAll()` method, we use Orbit’s `liveQuery()`. As you can probably guess, this queries for records of type 'message' to show on the index route.
+
+Next, let’s create the template for the route:
+
+`app/templates/index.hbs`:
+
+```hbs
+<form onsubmit={{action 'addMessage'}}>
+  {{input value=newMessage}}
+  <button type="submit">Add Message</button>
+</form>
+
+<ul>
+  {{#each model as |message|}}
+    <li>{{message.text}}</li>
+  {{/each}}
+</ul>
+```
+
+This template looks just like one we might have built in any Ember app. We create a form that binds a text value to the controller, and can be submitted to an action on the controller. The model we loaded in the route is an array of messages, which we display in a list.
+
+Now we need to implement the `addMessage` action on the controller.
+
+`app/controllers/index.js`:
+
+```js
+import Controller from '@ember/controller';
+
+export default Controller.extend({
+  actions: {
+    addMessage(e) {
+      e.preventDefault();
+      this.store.addRecord({ type: 'message', text: this.newMessage });
+      this.set('newMessage', '');
+    },
+  },
+});
+```
+
+Note that the way we access the model is also different: instead of Ember Data's `createRecord()`, we use Orbit's `addRecord()`. Note that we also include the type of the record in the object literal, rather than passing it as the first parameter.
+
+At this point, if you run your app, you should see that you can add messages to the list:
+
+SCREENSHOT
+
+But if you refresh the page, they're gone. We need to get back up to parity with Ember Data's features: saving and loading records via an API.
+
+One interesting thing to note: we’re going to add a lot of new files from here to incrementally improve the functionality of our data layer, but we won’t need to change the above files at all! This is the power of Orbit’s primitives: our business logic stays just the same (query the messages, add a message) but we change the configuration of what those concepts *mean*.
+
+## JSONAPISource and Push: Sending Updates to the Server
+
+To add features to our Orbit configuration, we’ll start with saving to the server. First, we'll define the remote data source that is our JSON API server:
+
+`app/data-sources/remote.js`:
+
+```js
+import JSONAPISource from '@orbit/jsonapi';
+import Orbit from '@orbit/data';
+import fetch from 'fetch';
+
+export default {
+  create(injections = {}) {
+    // Use `fetch` implementation from `ember-fetch`
+    Orbit.fetch = fetch;
+
+    injections.name = 'remote';
+    injections.host = 'http://localhost:3000';
+
+    return new JSONAPISource(injections);
+  }
+};
+```
+
+Note that we need to override `Orbit.fetch`; otherwise we'll get an error.
+
+QUESTION: WHY DOES BUILT-IN FETCH PRODUCE THE FOLLOWING ERROR? TypeError: Failed to execute 'fetch' on 'Window': Illegal invocation
```

Comment:
?

Reply:
There's an issue (and proposed fix) open, but no one's implemented the fix yet. It's trivial, so I'll just take care of it. See https://github.com/orbitjs/orbit/issues/452

Reply:
Cool, I'll mention that briefly in the post by way of explanation.